src/protocols/secure_channel/CASESession.cpp

```diff
@@ -974,140 +1074,143 @@ CHIP_ERROR CASESession::TryResumeSession(SessionResumptionStorage::ConstResumpti
+        // Next Step is to send Sigma2Resume message to the initiator
+        mNextStep = Step::kSendSigma2Resume;
 
-        // Early returning here, since we have sent Sigma2Resume, and no further processing is needed for the Sigma1 message
+        //  Early returning here, since the next Step is known to be Sigma2Resume, and no further processing is needed for the
+        //  Sigma1 message
         return CHIP_NO_ERROR;
     }
 
+    //  ParseSigma1 ensures that:
+    //  mRemotePubKey.Length() == initiatorPubKey.size() == kP256_PublicKey_Length.
+    memcpy(mRemotePubKey.Bytes(), parsedSigma1.initiatorEphPubKey.data(), mRemotePubKey.Length());
+
     // Attempt to match the initiator's desired destination based on local fabric table.
-    err = FindLocalNodeFromDestinationId(destinationIdentifier, initiatorRandom);
+    err = FindLocalNodeFromDestinationId(parsedSigma1.destinationId, parsedSigma1.initiatorRandom);
     if (err == CHIP_NO_ERROR)
     {
         ChipLogProgress(SecureChannel, "CASE matched destination ID: fabricIndex %u, NodeID 0x" ChipLogFormatX64,
                         static_cast<unsigned>(mFabricIndex), ChipLogValueX64(mLocalNodeId));
 
         // Side-effect of FindLocalNodeFromDestinationId success was that mFabricIndex/mLocalNodeId are now
         // set to the local fabric and associated NodeId that was targeted by the initiator.
+
+        mNextStep = Step::kSendSigma2;
     }
     else
     {
         ChipLogError(SecureChannel, "CASE failed to match destination ID with local fabrics");
-        ChipLogByteSpan(SecureChannel, destinationIdentifier);
-    }
-    SuccessOrExit(err);
+        ChipLogByteSpan(SecureChannel, parsedSigma1.destinationId);
 
-    // ParseSigma1 ensures that:
-    // mRemotePubKey.Length() == initiatorPubKey.size() == kP256_PublicKey_Length.
-    memcpy(mRemotePubKey.Bytes(), initiatorPubKey.data(), mRemotePubKey.Length());
+        // FindLocalNodeFromDestinationId returns CHIP_ERROR_KEY_NOT_FOUND if validation of DestinationID fails, which will trigger
+        // status Report with ProtocolCode = NoSharedTrustRoots.
+        mNextStep = Step::kSendStatusReport;
 
-    MATTER_LOG_METRIC_BEGIN(kMetricDeviceCASESessionSigma2);
-    err = SendSigma2();
-    if (CHIP_NO_ERROR != err)
-    {
-        MATTER_LOG_METRIC_END(kMetricDeviceCASESessionSigma2, err);
+        return err;
     }
-    SuccessOrExit(err);
-
-    mDelegate->OnSessionEstablishmentStarted();
-
-exit:
 
-    if (err == CHIP_ERROR_KEY_NOT_FOUND)
-    {
-        SendStatusReport(mExchangeCtxt, kProtocolCodeNoSharedRoot);
-        mState = State::kInitialized;
-    }
-    else if (err != CHIP_NO_ERROR)
-    {
-        SendStatusReport(mExchangeCtxt, kProtocolCodeInvalidParam);
-        mState = State::kInitialized;
-    }
-    return err;
+    return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR CASESession::SendSigma2Resume()
+CHIP_ERROR CASESession::PrepareSigma2Resume(EncodeSigma2ResInputs & output)
```

Comment:
Suggest rename of `output` to `sigma2ResInput`

Reply:
Done for this and PrepareSigma2() as well